### PR TITLE
Unsafe parsing

### LINKED ISSUE #16: [FEATURE REQUEST] `unsafe` parsing
**Description**
The keyword `unsafe` is not currently parsed in rust

**Motivation**
Unsafe is the other big half of the rust language, even if we are probably interest only in parsing the keyword for now.

### PR DESCRIPTION
**Pull request by simonegazza**
This PR aims to add the unsafe block during parsing of body of functions and as decorator for functions and methods.

Closes #16

### CODE REVIEW

File: rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		this.unsafe = unsafe;
+		this.location = location;
+	}
+
+	/**
+	 * Yields the decorator unsafe of this function.
+	 * 
+	 * @return true if the function is unsafe
+	 */
+	public boolean getUnsafe() {
```
Comment (VincenzoArceri) [inline rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java:181]: Rename in `isUnsafe`

**Review by VincenzoArceri — CHANGES_REQUESTED** [review]

File: rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java (hunk 1 above)
Comment (simonegazza) [reply rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java]: Fixed in commit 6962608144c610e50d0a1616902115f406f1dc57

File: rust-lisa/src/main/java/it/unipr/RustLiSA.java [hunk 2]
```diff
@@ -23,11 +29,16 @@ public class RustLiSA {
 	 * @throws IOException       if anything goes wrong during reading the file
 	 */
 	public static void main(String[] args) throws AnalysisException, IOException {
-		Program program = RustFrontend.processFile(args[0]);
+		Program program = RustFrontend.processFile("example.rs");
```
Comment (VincenzoArceri) [inline rust-lisa/src/main/java/it/unipr/RustLiSA.java:32]: Fix this.

File: rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+	}
+
+	/**
+	 * Yields a CFG from a descriptor.
+	 * 
+	 * @param descriptor the descriptor of this CFG
+	 * @param unsafe     the decorator unsafe
+	 * @param location   the source code location where this CFG is
+	 */
+	public RustCFG(CFGDescriptor descriptor, boolean unsafe, SourceCodeLocation location) {
```
Comment (VincenzoArceri) [inline rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java:191]: Move the construct at the beginning of the class.

File: rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+/**
+ * Implementation of a CFG of Rust.
+ * 
+ * @author <a href="mailto:vincenzo.arceri@unipr.it">Vincenzo Arceri</a>
+ * @author <a href="mailto:simone.gazza@studenti.unipr.it">Simone Gazza</a>
+ */
+public class RustCFG extends CFG {
+	private final boolean unsafe;
+	private final SourceCodeLocation location;
```
Comment (VincenzoArceri) [inline rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java:34]: Do we need `location` field? There's already a location in its `CFGDescriptor`.

File: rust-lisa/src/main/java/it/unipr/cfg/statement/RustUnsafeExitStatement.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import it.unive.lisa.util.datastructures.graph.GraphVisitor;
+
+/**
+ * Rust unsafe exiting statement.
+ * 
+ * @author <a href="mailto:vincenzo.arceri@unipr.it">Vincenzo Arceri</a>
+ * @author <a href="mailto:simone.gazza@studenti.unipr.it">Simone Gazza</a>
+ */
+public class RustUnsafeExitStatement extends Statement {
+
```
Comment (VincenzoArceri) [inline rust-lisa/src/main/java/it/unipr/cfg/statement/RustUnsafeExitStatement.java:24]: Missing reference to its `RustUnsafeEntryStatement`?

**Review by VincenzoArceri — CHANGES_REQUESTED** [review]

File: rust-lisa/src/main/java/it/unipr/RustLiSA.java (hunk 2 above)
Comment (simonegazza) [reply rust-lisa/src/main/java/it/unipr/RustLiSA.java]: Fixed in 393d7f1.

File: rust-lisa/src/main/java/it/unipr/cfg/statement/RustUnsafeExitStatement.java (hunk 5 above)
Comment (simonegazza) [reply rust-lisa/src/main/java/it/unipr/cfg/statement/RustUnsafeExitStatement.java:24]: Fixed in 4b5061a.

**Review by VincenzoArceri — APPROVED** [review]

**Pull request merged**